### PR TITLE
SQL & SCRIPTS: Kirin Updates

### PR DESCRIPTION
**Pull request by dr01d3r**
Kirin is now immune to sleep.

Kirin has reduced resistance to wind so elemental nukes are resisted
less often.

Kirin now uses Astral Flow and spawns a random avatar at 50-60% health.
An empty Astral_Flow.lua had to be added to support this.

Kirin's Avatar now has its own family system and mob skills to properly
handle the animation and prepare time for the blood pacts.

### CODE REVIEW

File: scripts/zones/The_Shrine_of_RuAvitau/mobs/Kirin_s_Avatar.lua [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+-----------------------------------
+-- onMobFight
+-- Actions:
+--  1 = Attacking
+-- 34 = Preparing Astral Flow
+-- 26 = Astral Flow used
+-----------------------------------
+function onMobFight( mob, target )
+    local id = mob:getID();
```
Comment (teschnei) [inline scripts/zones/The_Shrine_of_RuAvitau/mobs/Kirin_s_Avatar.lua]: can you do this onSpawn instead?

File: scripts/zones/The_Shrine_of_RuAvitau/mobs/Kirin_s_Avatar.lua [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+-----------------------------------
+function onMobDeath( mob, killer )
+end
+
+-----------------------------------
+-- OnMobDespawn
+-----------------------------------
+function onMobDespawn( mob )
+    -- Reset popped pet var..
+    mob:resetLocalVars();
```
Comment (teschnei) [inline scripts/zones/The_Shrine_of_RuAvitau/mobs/Kirin_s_Avatar.lua]: you don't need this, it's automatically wiped whenever he spawns

File: sql/mob_family_system.sql [hunk 3]
```diff
@@ -532,6 +532,7 @@ INSERT INTO `mob_family_system` VALUES (491,'Morimar',13,'Humanoid',0,40,90,110,
 INSERT INTO `mob_family_system` VALUES (492,'Balamor',9,'Demon',1,40,90,140,6,3,3,3,1,5,4,3,3,3,3,1,1,1,1,1,1,1,1,1,1,1,0.875,0);
 INSERT INTO `mob_family_system` VALUES (493,'Ashrakk',9,'Demon',1,40,90,140,6,3,3,3,1,5,4,3,3,3,3,1,1,1,1,1,1,1,1,1,1,1,0.875,0);
 INSERT INTO `mob_family_system` VALUES (494,'Sekhmet',6,'Beast',1,60,94,90,1,1,4,2,6,6,2,3,3,3,3,1,1,1,1,1,1,1,1.25,0.875,1,1,1,5);
+INSERT INTO `mob_family_system` VALUES (495,'Avatar-Kirin',5,'Avatar',0,40,100,120,3,3,3,3,3,3,3,3,3,3,3,1,1,1,1,1,1,1,1,1,1,1,1,0);
```
Comment (teschnei) [inline sql/mob_family_system.sql]: just name it Avatar or something, it doesn't need to be specific to kirin (other guys can use the same family later)

File: scripts/zones/The_Shrine_of_RuAvitau/mobs/Kirin_s_Avatar.lua (hunk 1 above)
Comment (dr01d3r) [inline scripts/zones/The_Shrine_of_RuAvitau/mobs/Kirin_s_Avatar.lua]: I believe I tried calling mob:useMobAbility in onMobSpawn (didn't work), but I will try once more to be absolutely sure.

Comment (teschnei) [inline scripts/zones/The_Shrine_of_RuAvitau/mobs/Kirin_s_Avatar.lua]: oh sorry, try onMobEngaged instead (the action queue is cleared when the mob engages)

Comment (dr01d3r) [inline scripts/zones/The_Shrine_of_RuAvitau/mobs/Kirin_s_Avatar.lua]: Might explain the hard crash I just received. :)  I will run again.

File: scripts/zones/The_Shrine_of_RuAvitau/mobs/Kirin_s_Avatar.lua (hunk 2 above)
Comment (dr01d3r) [inline scripts/zones/The_Shrine_of_RuAvitau/mobs/Kirin_s_Avatar.lua]: This was a remnant of the lua I copied to get started.  If it's an issue, I can remove it.

File: sql/mob_family_system.sql (hunk 3 above)
Comment (dr01d3r) [inline sql/mob_family_system.sql]: Would it make sense to call it something more specific to the astral flow ability?  There are already Avatar-Carbuncle, etc, that already exist.  I'm horrible with naming things, so I defer to you.  If "Avatar" is sufficient, I'll go with that.

Comment (TeoTwawki) [inline sql/mob_family_system.sql]: Suggestion: Astral_Flow_pet

Comment (dr01d3r) [inline sql/mob_family_system.sql]: I like it.

File: scripts/zones/The_Shrine_of_RuAvitau/mobs/Kirin_s_Avatar.lua (hunk 2 above)
Comment (teschnei) [inline scripts/zones/The_Shrine_of_RuAvitau/mobs/Kirin_s_Avatar.lua]: yep, remove it

**Comment by dr01d3r** [comment]
Last change request should be complete.

**Comment by teschnei** [comment]
and squash the commits and then i'll merge it

**Pull request merged**